test/test_eos_vinet.cpp

```diff
@@ -0,0 +1,491 @@
+//------------------------------------------------------------------------------
+// © 2021-2022. Triad National Security, LLC. All rights reserved.  This
+// program was produced under U.S. Government contract 89233218CNA000001
+// for Los Alamos National Laboratory (LANL), which is operated by Triad
+// National Security, LLC for the U.S.  Department of Energy/National
+// Nuclear Security Administration. All rights in the program are
+// reserved by Triad National Security, LLC, and the U.S. Department of
+// Energy/National Nuclear Security Administration. The Government is
+// granted for itself and others acting on its behalf a nonexclusive,
+// paid-up, irrevocable worldwide license in this material to reproduce,
+// prepare derivative works, distribute copies to the public, perform
+// publicly and display publicly, and to permit others to do so.
+//------------------------------------------------------------------------------
+
+#include <array>
+#include <cmath>
+#include <cstdio>
+#include <cstdlib>
+#include <limits>
+#ifndef CATCH_CONFIG_RUNNER
+#include "catch2/catch.hpp"
+#endif
+
+#include <singularity-eos/base/constants.hpp>
+#include <singularity-eos/eos/eos.hpp>
+#include <test/eos_unit_test_helpers.hpp>
+
+using singularity::EOS;
+using singularity::Vinet;
+
+SCENARIO("Vinet EOS rho sie", "[VectorEOS][VinetEOS]") {
```

Comment:
I didn't look too carefully at the tests, let's revisit when above discussions are resolved.